Classes/Controller/ModuleController.php

```diff
@@ -46,477 +56,418 @@ class ModuleController extends ActionController
+
+        $commandResult = $this->contentRepository->handle(
+            new RemoveNodeAggregate(
+                $liveWorkspace->currentContentStreamId,
+                $vocabularyNode->nodeAggregateId,
+                $vocabularyNode->originDimensionSpacePoint->toDimensionSpacePoint(),
+                NodeVariantSelectionStrategy::STRATEGY_ALL_VARIANTS
+            )
+        );
+        $commandResult->block();
+
+        $this->addFlashMessage(
+            sprintf('Deleted vocabulary %s', $vocabularyNode->getLabel())
+        );
+
+        $this->redirect('index', null, null, ['rootNodeAddress' => $this->nodeAddressFactory->createFromNode($rootNode)]);
     }
 
     /**
      * Show a form to create a new taxonomy
-     *
-     * @param NodeInterface $parent
-     * @return void
      */
-    public function newTaxonomyAction(NodeInterface $parent)
+    public function newTaxonomyAction(string $parentNodeAddress): void
     {
-        $flowQuery = new FlowQuery([$parent]);
-        $vocabulary = $flowQuery->closest('[instanceof ' . $this->taxonomyService->getVocabularyNodeType() . ']')->get(0);
-        $this->view->assign('vocabulary', $vocabulary);
-        $this->view->assign('parent', $parent);
+        $parentNode = $this->taxonomyService->getNodeByNodeAddress($parentNodeAddress);
+        $subgraph = $this->taxonomyService->getSubgraphForNode($parentNode);
+        $rootNode = $this->taxonomyService->findOrCreateRoot($subgraph);
+        $vocabularyNode = null;
+
+        if ($parentNode->nodeType->isOfType($this->taxonomyService->getTaxonomyNodeTypeName()->value)) {
+            $vocabularyNode = $this->taxonomyService->findVocabularyForNode($parentNode);
+        } elseif ($parentNode->nodeType->isOfType($this->taxonomyService->getVocabularyNodeTypeName()->value)) {
+            $vocabularyNode = $parentNode;
+        }
+
+        $this->view->assign('rootNode', $rootNode);
+        $this->view->assign('vocabularyNode', $vocabularyNode);
+        $this->view->assign('parentNode', $parentNode);
     }
 
     /**
      * Create a new taxonomy
-     *
-     * @param NodeInterface $parent
-     * @param array $properties
-     * @return void
+     * @param array<string, string> $properties
      */
-    public function createTaxonomyAction(NodeInterface $parent, array $properties)
+    public function createTaxonomyAction(string $parentNodeAddress, string $name, array $properties): void
     {
-        $taxonomyNodeType = $this->nodeTypeManager->getNodeType($this->taxonomyService->getTaxonomyNodeType());
-        $taxomonyProperties = $taxonomyNodeType->getProperties();
-
-        $nodeTemplate = new NodeTemplate();
-        $nodeTemplate->setNodeType($taxonomyNodeType);
-        $nodeTemplate->setName(CrUtitlity::renderValidNodeName($properties['title']));
+        $parentNode = $this->taxonomyService->getNodeByNodeAddress($parentNodeAddress);
+        $vocabularyNode = $this->taxonomyService->findVocabularyForNode($parentNode);
```

Comment:
This will fail if `$parentNode` is already a vocabulary.

```suggestion        
        $vocabularyNode =  $this->taxonomyService->getTaxonomyNodeTypeName()->equals($parentNode->nodeTypeName)
            ? $this->taxonomyService->findVocabularyForNode($parentNode)
            : $parentNode;
```

Reply:
Will check this again as I didn't need this while on the current neos dev collection

Reply:
Confirmed and adjusted. Thanks for reporting